modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -136,6 +138,18 @@ public CouchbaseContainer withBucket(final BucketDefinition bucketDefinition) {
         return this;
     }
 
+    /**
+     * Enables compatibility with community edition of couchbase server.
+     *
+     * @param communityMode if true, tries to be compatible with community edition.
+     * @return this {@link CouchbaseContainer} for chaining purposes.
+     */
+    public CouchbaseContainer withCommunityMode(final boolean communityMode) {
```

Comment:
technically, we should be able to infer this from the container type name? It would be convenient for the user to decide the best defaults based on the type used. .. so if you are okay to not have scope creep we could take the community bits out of this PR and iterate on it separately

Reply:
Yeah sure, sounds good. So something like, is "community" a substring of the tag name?

Reply:
indeed, they are prefixed with "community" (https://hub.docker.com/_/couchbase)

Reply:
But this won't work with custom images

Reply:
How about we have the default value be determined automatically based on the tag name prefix, but leave this method here so that the behaviour can be manually forced?

Still I'd agree with @daschl that it's probably better to tackle this in a separate PR, if this is in any way contentious.

Reply:
Actually, we might be able to read it from the container once started using the rest api .. I‘m not at the computer anymore but will check tomorrow

Reply:
Back at the computer. Since we do a lot with the rest api already anyways, I think we can solve this more elegantly. Once we have the server booted we can navigate to `/pools` and read the `isEnterprise` flag from the JSON response. No need to parse any custom strings etc. 

I'd still say let's put this in a follow-up change.